packages/node/src/utils/substrate.ts

```diff
@@ -163,25 +163,34 @@ export async function prefetchMetadata(
   await api.getBlockRegistry(hash);
 }
 
+/**
+ *
+ * @param api
+ * @param startHeight
+ * @param endHeight
+ * @param overallSpecVer exists if all blocks in the range have same parant specVersion
+ */
 export async function fetchBlocks(
   api: ApiPromise,
   startHeight: number,
   endHeight: number,
   overallSpecVer?: number,
 ): Promise<BlockContent[]> {
   const blocks = await fetchBlocksRange(api, startHeight, endHeight);
-  const blockHashs = blocks.map((b) => b.block.header.hash);
+  const parentBlockHashs = blocks.map((b) => b.block.header.parentHash);
   const [blockEvents, runtimeVersions] = await Promise.all([
-    fetchEventsRange(api, blockHashs),
-    overallSpecVer ? undefined : fetchRuntimeVersionRange(api, blockHashs),
+    fetchEventsRange(api, parentBlockHashs),
```

Comment:
if we use `parentBlockHashs` to fetch the events, then the events and blocks are miss matched

Reply:
good catch